src/engraving/dom/beambase.h

```diff
@@ -136,7 +190,13 @@ class BeamBase : public EngravingItem
     BeamBase(const ElementType& type, EngravingItem* parent, ElementFlags flags = ElementFlag::NOTHING);
     BeamBase(const BeamBase&);
 
+private:
```

Comment:
Instead of making these fields `private`, you could make (some of) them `protected`. This means that subclasses can access them directly, without the need to call a method. Advantages: might lead to slightly more compact code, and improved performance (although you'd hope that the compiler would optimise the method calls out anyway). 
It is true that encapsulating fields via methods is sometimes a good idea, to prevent unexpected modifications of these fields. However, in this case that reasoning doesn't really work, because the methods provide return non-const references, i.e. give exactly as much access to the fields as you would have when making the fields simply protected.